File: 06538.py
import sys


def main():
    for line in sys.stdin.readlines():
        line = line[:-1]
        idx = 0
        while idx < len(line):
            non_repeating = ""
            while idx + 1 < len(line) and line[idx] != line[idx + 1]:
                non_repeating += line[idx]
                idx += 1
            if idx + 1 == len(line):
                non_repeating += line[idx]
                idx += 1
            if non_repeating:
                sys.stdout.write("1")
                for letter in non_repeating:
                    if letter == "1":
                        sys.stdout.write("11")
                    else:
                        sys.stdout.write(letter)
                sys.stdout.write("1")
            if idx == len(line):
                break
            start = idx
            cnt = 0
            end = start
            while end < len(line) and line[end] == line[start] and cnt < 9:
                end += 1
                cnt += 1
            sys.stdout.write(f"{cnt}{line[idx]}")
            idx = end
        sys.stdout.write("\n")


main()

File: 06538.cpp
#include <iostream>
#include <string>

using namespace std;

int main(){
	ios_base::sync_with_stdio(false);
	cin.tie(NULL);
	string line;
	while(getline(cin, line)){
		unsigned idx = 0;
		while(idx < line.length()){
			unsigned start = idx, cnt = 0;
			unsigned end = start;
			while(end < line.length() && line[end] == line[start] && cnt < 9){
				++end;
				++cnt;
			}
			cout << cnt << line[idx];
			idx = end;
		}
		cout << '\n';
	}
	return 0;
}
